test/SILOptimizer/constant_evaluator_test.sil

```diff
@@ -0,0 +1,879 @@
+// RUN: %target-sil-opt -test-constant-evaluator %s 2>&1 | %FileCheck %s
```

Comment:
Nice, seems like this could eventually be used to simplify some of the `pound_assert.sil` tests that invoke flow-sensitive evaluation!

Reply:
Absolutely.